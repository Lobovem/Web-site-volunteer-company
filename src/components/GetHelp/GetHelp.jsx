import s from './GetHelp.module.scss';

export const GetHelp = () => {
  return (
    <div className={s.getHelp}>
      <div className={s.getHelp__contentWrap}>
        <div className={s.getHelp__imgWrap}>
          <img className={s.getHelp__img} src="./img/get-help.jpg" alt="get-help.jpg" />
        </div>
        <div className={s.getHelp__textWrap}>
          <h2 className={s.getHelp__title}>Хто може отримати допомогу від нашого фонду?</h2>

          <ul className={s.getHelp__descList}>
            <li className={s.getHelp__desc}>Військовослужбовці;</li>
            <li className={s.getHelp__desc}>Члени сімей військовослужбовців;</li>
            <li className={s.getHelp__desc}>Цивільні, які опинилися в скрутному становищі через військові дії;</li>
          </ul>

          <h2 className={s.getHelp__title}>Чим може допомогти наш фонд?</h2>

          <ul className={s.getHelp__descList}>
            <li className={s.getHelp__desc}>
              Медицини та здоров'я - медикаменти та все, що пов'язано зі здоров'ям та рятуванням наших захисників;
            </li>
            <li className={s.getHelp__desc}>Транспорт і засоби переміщення - авто для Сил оборони, включаючи ремонт та обслуговування ;</li>
            <li className={s.getHelp__desc}>Засоби розвідки - тепловізори, дрони, приціли і інші необхідні речі;</li>
            <li className={s.getHelp__desc}>"мокра" печатка;</li>
            <li className={s.getHelp__desc}>Загальна підтримка необхідним - продовольча підтримка, предмети одягу, засоби гігієни;</li>
            <li className={s.getHelp__desc}>Юридична підтримка;</li>
          </ul>

          <h2 className={s.getHelp__title}>Як отримати допомогу від нашого фонду?</h2>

          <p className={s.getHelp__desc}>
            Для того щоб записатись на отримання допомоги, та не вистоювати у чергах, потрібно заповнити реєстраційну форму. Вона
            автоматично реєструє заявку, та надсилає вам на пошту підтвердження реєстрації.{' '}
          </p>
        </div>
      </div>
    </div>
  );
};
