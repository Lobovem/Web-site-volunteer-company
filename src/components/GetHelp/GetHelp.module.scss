@import './src/styles/MixinsMedia.module.scss';
@import './src/styles/Variables.module.scss';
@import './src/styles/MixinsLayout.module.scss';
@import './src/styles/MixinsKit.module.scss';

.getHelp {

  &__contentWrap {
    display: flex;
    flex-direction: column;
    justify-content: center;

    @include lg-min {
      flex-direction: row;
      justify-content: space-between;
      gap: 40px;
    }
  }

  &__imgWrap {
    padding-bottom: 30px;
    width: 100%;

    @include lg-min {
      flex: 1;
      flex-basis: 40%;

    }

  }

  &__img {
    max-width: 100%;
    height: auto;
    border-radius: 15px;
    display: block;
    margin: 0 auto;
  }

  &__textWrap {
    @include lg-min {
      flex: 1;
      flex-basis: 60%;
    }
  }

  &__title {
    font-weight: $font-weight-500;
    font-size: $font-size-18;
    line-height: 18px;
    color: $color-black;
    mix-blend-mode: normal;
    margin: 25px 0;
  }

  &__descList {
    list-style-type: disc;
    margin-left: 20px;
    margin-bottom: 15px;
  }

  &__desc {
    font-weight: $font-weight-400;
    font-size: $font-size-14;
    line-height: 21px;
    color: $color-black;
    mix-blend-mode: normal;
    margin-bottom: 10px;
  }
}