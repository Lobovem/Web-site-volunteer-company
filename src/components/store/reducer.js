import { createSlice } from '@reduxjs/toolkit';

const initialState = {
  burgerMenuState: false,
};

export const appReducer = createSlice({
  name: 'app',
  initialState: initialState,
  reducers: {
    isOpenBurgerMenu: (state, action) => {
      state.burgerMenuState = !state.burgerMenuState;
    },
  },
});

export const burgerMenuSelector = (state) => state.app.burgerMenuState;

export const { isOpenBurgerMenu } = appReducer.actions;
export default appReducer.reducer;
