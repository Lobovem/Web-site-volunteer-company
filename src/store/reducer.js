import { createSlice } from '@reduxjs/toolkit';

const initialState = {
  todoListState: [],
};

export const appReducer = createSlice({
  name: 'todoList',
  initialState: initialState,
  reducers: {
    // add_todo: (state, action) => {
    //   state.todoListState = [
    //     ...state.todoListState,
    //     {
    //       id: Math.random(),
    //       task: action.payload,
    //       complete: false,
    //       isEditing: false,
    //     },
    //   ];
    // },
    // delete_todo: (state, action) => {
    //   state.todoListState = state.todoListState.filter((todo) => todo.id !== action.payload.id);
    // },
  },
});

// export const userInputSelector = (state) => state.todoList.inputState;
// export const todoListSelector = (state) => state.todoList.todoListState;

// export const { add_todo } = todoReducer.actions;
export default appReducer.reducer;
