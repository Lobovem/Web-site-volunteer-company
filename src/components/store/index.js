// import configureStore from '@reduxjs/toolkit';
import { configureStore } from '@reduxjs/toolkit';
import { appReducer } from './reducer';

export const store = configureStore({
  reducer: {
    todoList: appReducer.reducer,
  },
});
