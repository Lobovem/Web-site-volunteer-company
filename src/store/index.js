// import configureStore from '@reduxjs/toolkit';
import { configureStore } from '@reduxjs/toolkit';
import { todoReducer } from './reducer';

export const store = configureStore({
  reducer: {
    todoList: todoReducer.reducer,
  },
});
