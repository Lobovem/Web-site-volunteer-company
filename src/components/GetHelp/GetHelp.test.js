import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { GetHelp } from './GetHelp';
import { Provider } from 'react-redux';
import { store } from '../../redux/config/store';

describe('GetHelp` component', () => {
  test('renders GetHelp component', () => {
    // eslint-disable-next-line testing-library/render-result-naming-convention
    const component = render(
      <Provider store={store}>
        <Router>
          <GetHelp />
        </Router>
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });

  test('should render the titles (H2) component GetHelp', () => {
    render(
      <Provider store={store}>
        <Router>
          <GetHelp />
        </Router>
      </Provider>
    );
    const titleSection = screen.getAllByRole('heading', { level: 2 });
    expect(titleSection).toHaveLength(3);
  });

  test('renders and search GetHelp component with 1 images', () => {
    render(
      <Provider store={store}>
        <Router>
          <GetHelp />
        </Router>
      </Provider>
    );
    const images = screen.getAllByRole('img');
    expect(images).toHaveLength(1);
  });
});
