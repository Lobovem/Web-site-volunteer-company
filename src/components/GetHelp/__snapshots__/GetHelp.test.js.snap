// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetHelp\` component renders GetHelp component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="getHelp"
      >
        <div
          class="getHelp__contentWrap"
        >
          <div
            class="getHelp__imgWrap"
          >
            <img
              alt="get-help.jpg"
              class="getHelp__img"
              src="/img/get-help.jpg"
            />
          </div>
          <div
            class="getHelp__textWrap"
          >
            <h2
              class="getHelp__title"
            >
              Хто може отримати допомогу від нашого фонду?
            </h2>
            <ul
              class="getHelp__descList"
            >
              <li
                class="getHelp__desc"
              >
                Військовослужбовці;
              </li>
              <li
                class="getHelp__desc"
              >
                Члени сімей військовослужбовців;
              </li>
              <li
                class="getHelp__desc"
              >
                Цивільні, які опинилися в скрутному становищі через військові дії;
              </li>
            </ul>
            <h2
              class="getHelp__title"
            >
              Чим може допомогти наш фонд?
            </h2>
            <ul
              class="getHelp__descList"
            >
              <li
                class="getHelp__desc"
              >
                Медицини та здоров'я - медикаменти та все, що пов'язано зі здоров'ям та рятуванням наших захисників;
              </li>
              <li
                class="getHelp__desc"
              >
                Транспорт і засоби переміщення - авто для Сил оборони, включаючи ремонт та обслуговування ;
              </li>
              <li
                class="getHelp__desc"
              >
                Засоби розвідки - тепловізори, дрони, приціли і інші необхідні речі;
              </li>
              <li
                class="getHelp__desc"
              >
                "мокра" печатка;
              </li>
              <li
                class="getHelp__desc"
              >
                Загальна підтримка необхідним - продовольча підтримка, предмети одягу, засоби гігієни;
              </li>
              <li
                class="getHelp__desc"
              >
                Юридична підтримка;
              </li>
            </ul>
            <h2
              class="getHelp__title"
            >
              Як отримати допомогу від нашого фонду?
            </h2>
            <p
              class="getHelp__desc"
            >
              Для того щоб записатись на отримання допомоги, та не вистоювати у чергах, потрібно заповнити реєстраційну форму. Вона автоматично реєструє заявку, та надсилає вам на пошту підтвердження реєстрації.
               
            </p>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="getHelp"
    >
      <div
        class="getHelp__contentWrap"
      >
        <div
          class="getHelp__imgWrap"
        >
          <img
            alt="get-help.jpg"
            class="getHelp__img"
            src="/img/get-help.jpg"
          />
        </div>
        <div
          class="getHelp__textWrap"
        >
          <h2
            class="getHelp__title"
          >
            Хто може отримати допомогу від нашого фонду?
          </h2>
          <ul
            class="getHelp__descList"
          >
            <li
              class="getHelp__desc"
            >
              Військовослужбовці;
            </li>
            <li
              class="getHelp__desc"
            >
              Члени сімей військовослужбовців;
            </li>
            <li
              class="getHelp__desc"
            >
              Цивільні, які опинилися в скрутному становищі через військові дії;
            </li>
          </ul>
          <h2
            class="getHelp__title"
          >
            Чим може допомогти наш фонд?
          </h2>
          <ul
            class="getHelp__descList"
          >
            <li
              class="getHelp__desc"
            >
              Медицини та здоров'я - медикаменти та все, що пов'язано зі здоров'ям та рятуванням наших захисників;
            </li>
            <li
              class="getHelp__desc"
            >
              Транспорт і засоби переміщення - авто для Сил оборони, включаючи ремонт та обслуговування ;
            </li>
            <li
              class="getHelp__desc"
            >
              Засоби розвідки - тепловізори, дрони, приціли і інші необхідні речі;
            </li>
            <li
              class="getHelp__desc"
            >
              "мокра" печатка;
            </li>
            <li
              class="getHelp__desc"
            >
              Загальна підтримка необхідним - продовольча підтримка, предмети одягу, засоби гігієни;
            </li>
            <li
              class="getHelp__desc"
            >
              Юридична підтримка;
            </li>
          </ul>
          <h2
            class="getHelp__title"
          >
            Як отримати допомогу від нашого фонду?
          </h2>
          <p
            class="getHelp__desc"
          >
            Для того щоб записатись на отримання допомоги, та не вистоювати у чергах, потрібно заповнити реєстраційну форму. Вона автоматично реєструє заявку, та надсилає вам на пошту підтвердження реєстрації.
             
          </p>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
